org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/ProjectOpenedOrClosedListener.java

```diff
@@ -263,15 +263,22 @@ protected void processClosedProjects(IProgressMonitor monitor) {
 		if (task.isEmpty()) {
 			return;
 		}
-		String projectNames = Joiner.on(", ").join(task.getProjectNames());
+		ImmutableSet<String> removedProjects = task.getProjectNames();
+		String projectNames = Joiner.on(", ").join(removedProjects);
 		String taskName = Messages.ProjectOpenedOrClosedListener_RemovingProject + projectNames
 				+ Messages.ProjectOpenedOrClosedListener_FromIndex;
 		monitor.setTaskName(taskName);
 		SubMonitor progress = SubMonitor.convert(monitor, 1);
 		try {
 			ResourceSet resourceSet = createResourceSet();
-			BuildData buildData = new BuildData(null, resourceSet, task.getToBeBuilt(), queuedBuildData, false,
-					BuildManagerAccess::requestBuild);
+			BuildData buildData = new BuildData(
+					"",
```

Comment:
null?

Reply:
If you use `workspace.getProject(buildData.getProjectName())` you'd get an assertion error. With the empty string, you'll find a project that does not exist. We do have a test that fails if we use null here. I'm not really passionate - changing it into one or the other direction would be ok'ish. But the empty string is more failsafe I suppose.

Reply:
ok. can you please add a javadoc comment to BuildData.getProjectName